File: 프로그래머스/Lv.1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
from collections import defaultdict
def solution(participant, completion):
    part_dict = defaultdict(int)
    for p in participant:
        part_dict[p] += 1
    for c in completion:
        part_dict[c] -= 1
        if part_dict[c] == 0:
            part_dict.pop(c)
    for i in part_dict:
        return i

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;
import static java.util.stream.Collectors.*;
class Solution {
    public String solution(String[] participant, String[] completion) {
        
        String answer = "";
        Map<String, Long> comMap = Arrays.stream(completion)
                                            .collect(
                                            groupingBy(d -> d, counting()));
        
        for (String par : participant) {
            if (!comMap.containsKey(par)) {
                answer = par;
                break;
            }
            comMap.put(par, comMap.get(par)-1);
            if (comMap.get(par) == 0) 
                comMap.remove(par);
        }
        return answer;
    }
}